packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelFormFields/DefaultValueField.tsx

```diff
@@ -115,8 +124,15 @@ function DefaultValueListField(
     <MultiValueField
       value={value}
       onChange={arr => {
-        // Convert null to undefined
-        onChange(arr || undefined)
+        if (arr == null) {
+          // Convert null to undefined
+          onChange(undefined)
+        } else if (columnModel.columnType === ColumnTypeEnum.DATE_LIST) {
+          onChange(arr.map(val => dayjs(val).valueOf()))
+        } else {
+          // Convert null to undefined
```

Comment:
Looks like this comment may not apply here, since this conversion is handled above